### PR TITLE
refactor(cli/build): move to using common processjstask

### PR DESCRIPTION
**Pull request by ewanharris**
This aligns Windows with the same js processing code as Android and iOS, which handles all the knowledge like dont minify the internal common files, incremental builds etc

Fixes [TIMOB-27452](https://jira.appcelerator.org/browse/TIMOB-27452)

Pinging @janvennemann to just make sure I'm using the processjs stuff right

### CODE REVIEW

File: cli/commands/_build/copy.js [hunk 1]
```diff
@@ -543,133 +544,117 @@ function copyResources(next) {
...
-				// ignore appxmanifest
-				if (prop !== 'manifest') {
-					props[prop] = this.tiapp.windows[prop];
-				}
-			}, this);
-			fs.writeFileSync(
-				appPropsFile,
-				JSON.stringify(props)
-			);
-			this.encryptJS && jsFilesToEncrypt.push('_app_props_.json');
```
Comment (sgtcoolguy) [inline cli/commands/_build/copy.js]: My understanding is that each platform is still doing the app props/info files separately and it is not actually part of the process-js-task code right now.

File: cli/commands/_build/copy.js [hunk 2]
```diff
@@ -543,133 +544,117 @@ function copyResources(next) {
...
 			}
-
-			this.processEncryption(next);
 		});
+
+		task.run()
+			.then(() => {
+
+				const tasks = copyUnmodified.map(relPath => {
+					const from = jsFiles[relPath];
```
Comment (ewanharris) [inline cli/commands/_build/copy.js]: This code makes no sense I think, will fix

**Comment by sgtcoolguy** [comment]
I mistakenly thought the app_props and info json file stuff was removed, but it was just re-arranged. But that does raise the obvious question to @janvennemann (and the rest of us) - can we move more of this common code into the process-js-task (or some common build code location) - because it looks like each platform is doing it's own _app_props_.json stuff.

The _app_info_.json seems specific to Windows.

File: cli/commands/_build/copy.js (hunk 2 above)
Comment (ewanharris) [reply cli/commands/_build/copy.js]: Confirmed non-sensical code, fix incoming

Comment (ewanharris) [reply cli/commands/_build/copy.js]: Fix in 1cb9de77fe5088d9e8f5ea4a4853af4825980780

File: cli/commands/_build/copy.js [hunk 3]
```diff
@@ -543,133 +544,119 @@ function copyResources(next) {
...
-				// if we're encrypting the JavaScript, copy the files to the assets dir
-				// for processing later
-				if (this.encryptJS) {
-					to = path.join(this.buildTargetAssetsDir, id);
-					jsFilesToEncrypt.push(id);
-				}
-
-				try {
-					this.cli.createHook('build.windows.analyzeJsFile', this, function (from, to, cb) {
-						this.cli.createHook('build.windows.copyResource', this, function (from, to, cb) {
```
[the remark this answers is not in the document]
Comment (ewanharris) [reply cli/commands/_build/copy.js]: I think we'll have to update the ProcessJsTask to do that, will put a PR on titanium_mobile

[the remark this answers is not in the document]
Comment (infosia) [reply cli/commands/_build/copy.js]: I think that makes sense. I'll merge this PR.

**Comment by infosia** [comment]
Merged, still needs to update the ProcessJsTask to do that, @ewanharris will put a PR on titanium_mobile

**Pull request merged**